lib/ui/screens/types/modal_contents.dart

```diff
@@ -16,18 +25,138 @@ class ModalContents extends StatelessWidget {
   final int index;
   final double width;
   final ScrollController scroller;
+
+  @override
+  State<ModalContents> createState() => _ModalContentsState();
+}
+
+class _ModalContentsState extends State<ModalContents> {
+  final List<bool> _isOpen = [false, false, false];
+
+  void _onPokemonPress(int index, Pokemon pokemon) {
+    context.read(currentPokemonStateProvider).setPokemon(index, pokemon);
+
+    print(pokemon.name);
+
+    AppNavigator.push(Routes.pokemonInfo, pokemon);
+  }
+
+  ExpansionPanel _buildTypePokemonPanel(List<Pokemon> pokemons) {
+    final pokemonsFiltered = pokemons
+        .where((pokemon) => pokemon.types.contains(types[widget.index].type))
+        .toList();
+    return ExpansionPanel(
+      headerBuilder: (context, isOpen) {
+        return Row(
+          children: [
+            Padding(
+              padding: const EdgeInsets.only(left: 8.0),
+              child: Image(
+                image: AppImages.pokeball,
+                width: 30,
+                height: 30,
+                color: types[widget.index].color.withOpacity(0.5),
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.only(left: 8.0),
+              child: Text(
+                  "${getEnumValue(types[widget.index].type)[0].toUpperCase() + getEnumValue(types[widget.index].type).substring(1)} Type " +
+                      "Pokemons"),
+            )
+          ],
+        );
+      },
+      canTapOnHeader: true,
+      body: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 10.0),
+        child: GridView.count(
+          shrinkWrap: true,
+          crossAxisCount: 2,
+          childAspectRatio: 1.4,
+          crossAxisSpacing: 10,
+          mainAxisSpacing: 10,
+          physics: const NeverScrollableScrollPhysics(),
+          children: pokemonsFiltered.map((pokemon) {
+            return PokemonCard(
+              pokemon,
+              index: pokemons.indexOf(pokemon),
+              onPress: () => _onPokemonPress(pokemons.indexOf(pokemon), pokemon),
+            );
+          }).toList(),
+        ),
+      ),
+      isExpanded: _isOpen[0],
+    );
+  }
+
+  ExpansionPanel _buildTypeItemsPanel() => ExpansionPanel(
```

Comment:
You should use this syntax instead so that we can easily add any extra logic if needed, and it's easier for the eyes :D

```dart
<Type> _buildSomething() {
    return <widget>
}
```